apidocs.yaml

```diff
@@ -0,0 +1,826 @@
+  - name: members
+    description: All endpoints related to members
+  - name: terms
+    description: All endpoints related to terms
+  - name: datasets
+    description: All endpoints related to datasets
+  - name: forms
+    description: All endpoints related to forms
+  - name: actions
+    description: All endpoints related to user actions
+paths:
+  /api/v1/applications:
+    get:
+      summary: List applications
+      operationId: list_applications_v1
+      tags:
+        - "applications"
+      responses:
+        "200":
+          description: Successful Response
+          content:
+            application/json:
+              schema:
+                type: array
+                title: collection of applications
+                items:
+                  $ref: "#/components/schemas/ListedApplication"
+      security:
+        - auth:
+          - read:applications
+  /api/v1/applications/{id}:
+    get:
+      summary: Retrieve application
+      operationId: retrieve_application_v1
+      tags:
+        - "applications"
+      parameters:
+        - name: id
+          in: path
+          description: ID of application to retrieve
+          required: true
+          schema:
+            type: string
+      responses:
+        "200":
+          description: Successful Response
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/RetrievedApplication"
+      security:
+        - auth:
+          - read:applications
+  /api/v1/applications/{id}/save-forms-and-duos:
+    post:
+      summary: save application forms and duos
+      operationId: save_application_forms_and_duos_v1
+      tags:
+        - "forms"
+      parameters:
+        - name: id
+          in: path
+          description: ID of application to update
+          required: true
+          schema:
+            type: string
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/SaveFormsAndDuos"
+      responses:
+        "204":
+          description: Successful Response (no content)
+      security:
+        - auth:
+          - write:applications
+  /api/v1/applications/{id}/submit:
+    post:
+      summary: Submit application
+      operationId: submit_application_v1
+      tags:
+        - "actions"
+      parameters:
+        - name: id
+          in: path
+          description: ID of application to submit
+          required: true
+          schema:
+            type: string
+      responses:
+        "204":
+          description: Successful Response (no content)
+        "422":
+          description: Validation warnings
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: "#/components/schemas/ValidationWarnings"
+      security:
+        - auth:
+          - write:applications
+  /api/v1/applications/create-application:
```

Comment:
isnt just having POST `/api/v1/applications` more "restful" than POST `/api/v1/applications/create-application` for creation?

Reply:
It would make sense if we were posting an application, but this endpoint creates an application from a list of selected datasets.

What will do behind the seen is fetching from DAAM the configuration of an application (forms and their fields, licenses, duo, etc...), fill a object of type Application and assign it to the datasets.